### ISSUE TITLE
Error when choosing Download to CSV while running a query

### ISSUE THREAD

**Issue by tdressel** [ozone-development/meridian#302]
STR
DEVELOP branch
Run a big query
Before query is finished, choose Download to CSV
RESULTS - Green toastr notification - "Query initiated" and Red toastr notification - "Connection to server failed."
EXPECTED RESULTS - Should open new browser tab and display message saying ,  "no data to be displayed"

**Comment by cwilmsen** [comment]
This works for me with the new export stuff. (No branch)

**Comment by tdressel** [comment]
This is still an issue on develop/local branch.

**Comment by cwilmsen** [comment]
Still don't see an issue. It works for me. Do you have any more specifics if it still isn't working for you?
I am running a really big mock query on Chrome, but the client shouldn't matter.

**Comment by tdressel** [comment]
I was able to repro this again today on develop branch. ONLY fails on FF.

**Comment by jlfoster2** [comment]
@cwilmsen ...try this is any version of FF and let me know if you get that error.  I think this is an extreme edge case. Also think we need to consider not allowing features to be exported until a query is complete.  Thoughts on that?

**Comment by cwilmsen** [comment]
It kind of looks like FF is 'changing pages' temporarily during a file download. This causes all in-progress calls to be aborted. When you have a dev console up, when you start the download, everything resets. The pages keeps it's data state, but shuts everything else down.

We are currently using window.location.assign to start the download. There is another option, window.open. This causes the download to start in another window. I have shown that doing this does indeed not stop the query. It also causes geoJSON and KML files to fail. Plus, it looks bad.

The problem is that opening a new page does not set up everything that we need to interact with the server, like a sessionId and possible a cert. Three options: make downloads not need any verification (not going to happen), not allow downloads to happen when ANY calls are in progress, or enforce Chrome.

Basically, not allowing a download while any call is running is pretty much the only option, unless someone can find another. The issue quickly becomes that some calls take a VERY long time to finish.

Thoughts?

**Comment by cwilmsen** [comment]
Anyone?

**Comment by cwilmsen** [comment]
@jlfoster2 You closed it without any comment. What do you think?

**Comment by tdressel** [comment]
same behavior as reported on develop
